aeppl/convolutions.py

```diff
@@ -0,0 +1,56 @@
+import aesara
+import aesara.tensor as at
+from aesara.graph.rewriting.basic import EquilibriumGraphRewriter, node_rewriter
+from aesara.scalar.basic import Add
+from aesara.tensor.elemwise import Elemwise
+from aesara.tensor.random.basic import NormalRV, normal
+
+from aeppl.rewriting import logprob_rewrites_db
+
+
+@node_rewriter((Elemwise,))
+def add_independent_normals(fgraph, node):
+    if not isinstance(node.op.scalar_op, Add):
+        return None
+
+    X_rv, Y_rv = node.inputs
+
+    if not (X_rv.owner and Y_rv.owner) or not (
+        isinstance(X_rv.owner.op, NormalRV) and isinstance(Y_rv.owner.op, NormalRV)
+    ):
+        return None
+
+    old_rv = node.outputs[0]
+
+    mu_x, sigma_x, mu_y, sigma_y, _ = at.broadcast_arrays(
+        *(X_rv.owner.inputs[-2:] + Y_rv.owner.inputs[-2:] + [old_rv])
+    )
+
+    new_rng = X_rv.owner.inputs[0]
```

Comment:
In one of @brandonwillard's comments in another discussion, we wrote
```python
new_rng = X_rv.owner.inputs[0].clone()
```
However, by cloning this `new_rng` for the `normal.make_node(...)` below, I obtain the following error:
```python
aesara.graph.utils.MissingInputError: Input 0 (RandomGeneratorSharedVariable(<Generator(PCG64) at 
0x7FF373115460>)) of the graph (indices start from 0), used to compute normal_rv{0, (0, 0), floatX, False}
(RandomGeneratorSharedVariable(<Generator(PCG64) at 0x7FF373115460>), TensorConstant{[5 1]}, 
TensorConstant{11}, Elemwise{add,no_inplace}.0, Elemwise{sqrt,no_inplace}.0), was not provided and not given a 
value. Use the Aesara flag exception_verbosity='high', for more information on this error.
```
I didn't spend that much time on this bug specifically, but I'm not sure why cloning the RNG yields a `MissingInputError`...

Reply:
Ah, yeah, you're seeing that error because `FunctionGraph` is really strict about adding new input variables (i.e. `Variable`s that have no/`None` `Variable.owner` values and are not `Constant`s) by default.  One quick way around this is to add the statement `fgraph.add_input(new_rng)`.

In the long run, we might want to make the `SharedVariable`s&mdash;or at least `RandomTypeSharedVariable`s&mdash;work like `Constant`s in this situation.  In general rewriting (e.g. for compilation purposes), that might not be the best idea, though, especially if there are `SharedVariable.default_update`s involved, because the associated updates won't be picked up during the regular `aesara.function` compilation process.

Reply:
You should be able to add `fgraph.add_input(new_rng)` to the rewrite, since `fgraph` is one of the arguments.

Reply:
Ah, of course... I just forced pushed again